core/app/models/spree/promotion/rules/taxon.rb

```diff
@@ -62,7 +62,7 @@ def taxon_ids_string=(s)
         private
 
         def warn_invalid_match_policy(assume:)
-          ActiveSupport::Deprecation.warn(
+          Spree::Deprecation.warn(
             "#{self.class.name} id=#{id} has unexpected match policy #{preferred_match_policy.inspect}. "\
             "Interpreting it as '#{assume}'." \
             "In future versions of Solidus this will be an error."
```

Comment:
This line could go, as `Spree::Deprecation` specifies a version where this errors.